.github/workflows/CI.yml

```diff
@@ -159,6 +159,39 @@ jobs:
         command: test
         args: --all
 
+  test-build-wasm:
+    needs: check
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        # TODO(securityinsanity): slowly add wasm32 test runner to each crate, and move to seperate actions that run tests.
+        subcrate:
+        - tracing
+        - tracing-appender
+        - tracing-attributes
+        - tracing-core
+        - tracing-error
+        - tracing-flame
+        - tracing-journald
+        - tracing-log
+        - tracing-macros
+        - tracing-opentelemetry
+        - tracing-serde
+        - tracing-subscriber
+        - tracing-tower
+    steps:
+    - uses: actions/checkout@main
+    - uses: actions-rs/toolchain@v1
+      with:
+        target: wasm32-unknown-unknown
+        toolchain: stable
+        override: true
+    - name: build all tests
+      uses: actions-rs/cargo@v1
+      with:
+        command: test
+        args: --no-run -p ${{ matrix.subcrate }}
```

Comment:
Take it or leave it: I think we can also use `cargo check --tests` to build but not run tests. But, `test --no-run` might be nicer just because when we actually enable tests, we can just remove the `--no-run` :)

Reply:
Didn't actually know that, but good to know now! I don't particularly ha e a preference so if we're okay with this we can just leave it.